sdk/src/main/java/io/radar/sdk/Radar.kt

```diff
@@ -2225,6 +2246,175 @@ object Radar {
+     * @param[metadata] The metadata associated with the event.
+     * @param[block] A block callback
+     */
+    @JvmStatic
+    fun sendEvent(
+        customType: String,
+        metadata: JSONObject?,
+        block: (status: RadarStatus, location: Location?, events: Array<RadarEvent>?, user: RadarUser?) -> Unit
+    ) {
+        sendEvent(customType, metadata, object : RadarSendEventCallback {
+            override fun onComplete(
+                status: RadarStatus,
+                location: Location?,
+                events: Array<RadarEvent>?,
+                user: RadarUser?
+            ) {
+                block(status, location, events, user)
+            }
+        })
+    }
+
+    /**
+     * Sends a custom event with a manually provided location.
+     *
+     * @see [](https://radar.com/documentation/api#send-a-custom-event)
+     *
+     * @param[customType] The name of the event.
+     * @param[location] The location of the event.
+     * @param[metadata] The metadata associated with the event.
+     * @param[callback] A callback.
+     */
+    @JvmStatic
+    fun sendEvent(
+        customType: String,
+        location: Location,
+        metadata: JSONObject?,
+        callback: RadarSendEventCallback
+    ) {
+        if (!initialized) {
+            callback.onComplete(RadarStatus.ERROR_PUBLISHABLE_KEY)
+
+            return
+        }
+
+        trackOnce(location, object : RadarTrackCallback {
+            override fun onComplete(status: RadarStatus, location: Location?, events: Array<RadarEvent>?, user: RadarUser?) {
+                if (status != RadarStatus.SUCCESS || location == null) {
+                    handler.post {
+                        callback.onComplete(status)
+                    }
+
+                    return
+                }
+
+                sendEvent(customType, metadata, location, events, user, callback)
+            }
+        })
+    }
+
+    /**
+     * Sends a custom event with a manually provided location.
+     *
+     * @see [](https://radar.com/documentation/api#send-a-custom-event)
+     *
+     * @param[customType] The name of the event.
+     * @param[location] The location of the event.
+     * @param[metadata] The metadata associated with the event.
+     * @param[block] A block callback.
+     */
+    @JvmStatic
+    fun sendEvent(
+        customType: String,
+        location: Location,
+        metadata: JSONObject?,
+        block: (status: RadarStatus, location: Location?, events: Array<RadarEvent>?, user: RadarUser?) -> Unit
+    ) {
+        sendEvent(customType, location, metadata, object : RadarSendEventCallback {
+            override fun onComplete(
+                status: RadarStatus,
+                location: Location?,
+                events: Array<RadarEvent>?,
+                user: RadarUser?
+            ) {
+                block(status, location, events, user)
+            }
+        })
+    }
+
+    @JvmStatic
+    private fun sendEvent(
```

Comment:
agree with the decision to refactor this logic into its own function, but technically it's still not happening in the API client - could this happen in `apiClient.sendEvent` passing through the `events` arg from `/track`?

Reply:
See my comment on lines 998-1001 for why it's here, not there.